ouroboros-consensus/test/consensus-test/Test/Consensus/Mempool.hs

```diff
@@ -492,6 +508,7 @@ genValidTxs = go []
           (tx, ledger') <- genValidTx ledger
           go (tx:txs) (n - 1) ledger'
 
+-- | Generate a valid transaction (but ignoring any per-tx size limits).
```

Comment:
This comment is a bit confusing. Does `mustBeValid` not check the per-tx size limit?

Reply:
It does not as we use `testLedgerConfigNoSizeLimits` in this particular case. The current approach is to first generate "non-large" txs, and then use that to set the mempool size (already before this PR) and the per-tx limit.

Reply:
We had a call. It all looks good. Alex is going to add a GHC-style `Note` explaining how the piece interact (ie the new config argument).

Reply:
Added the following note:
```
Note [Transaction size limit]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

An important property of the mempool is that adding a transaction that can never
fit into the mempool must not block, also see
https://github.com/IntersectMBO/ouroboros-consensus/issues/1226. We test this
while generating a TestSetupWithTxs by always including a transaction that is
larger than the entire mempool, and setting the per-tx size limit such that just
this transaction is invalid due to its size, but not impacting the validity of
any other transactions. Therefore, we disable the size limit in e.g.
'genValidTx' to only capture UTxO-related validity for them by using an
appropriate ledger config ('testLedgerConfigNoSizeLimits').
```